griptape/common/prompt_stack/prompt_stack.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+from attrs import define, field
+
+from griptape.artifacts import TextArtifact, BaseArtifact, ListArtifact, ImageArtifact
+from griptape.mixins import SerializableMixin
+from griptape.common import PromptStackMessage, TextPromptStackContent, BasePromptStackContent, ImagePromptStackContent
+
+
+@define
+class PromptStack(SerializableMixin):
```

Comment:
I like the shortening of Content/Message. Also in favor of renaming Prompt Stack since it's grown to be much more than just a list of string prompts. Let's try `Messages` and see how it sits.

Reply:
Why `Stack`?

Reply:
I was trying to understand the origins of `Stack` in the context of a prompt stack. I found this article: [Unveiling the Mastery of Prompt Stacking](https://www.linkedin.com/pulse/unveiling-mastery-prompt-stacking-your-roadmap-precision-mike-meehan), which defines Prompt Stacking as "meticulously crafting a series of prompts to guide the model step by step towards a desired outcome." with the following basic example:
- Basic Prompt: “Translate the following text to French: {text}”
- Stacked Prompt: “You are a proficient bilingual assistant. Translate the following text to French: {text}”

So it sounds like prompt stacking is more or less synonymous with prompt engineering. 

This makes me realize that stack is definitely a common terminology wrt LLMs, but that it doesn't seem to directly align with our `PromptStack`, since it would probably be tacking place within our prompt templates (I suppose an individual "prompt" roughly equals a statement or instruction or rule within the template). `MessageStack` does work, since it is "stacking" messages just as much as "stacking" prompts does, though the computer science pupil in me is screaming "but it's not a stack!" (it's ok we can just ignore him).

So long story short, I'm fine with `MessageStack`, though I'd still prefer something that no one would ever need to ask "why" about.